src/datautils/surfacedata.ts

```diff
@@ -180,38 +180,39 @@ const unassignedColorScale = scaleOrdinal<number, string>()
 function findBestMatchingBaseIndex(top: any, index: number, surfaces: any): number {
   const nextIndex: number = index + 1;
 
-  if (!surfaces || nextIndex >= surfaces.length) return null;
+  if (!surfaces || nextIndex >= surfaces.length) {
+    return null;
+  }
 
   // if there is a matching base by name, use that
   let candidate: any = surfaces[nextIndex];
   if (candidate.isBase && candidate.name === top.name) {
     return nextIndex;
   }
-  // verify that each data point in the top has a corresponding
-  // data point in the base candidate. If not, find another candidate
-  // if none match up, select the best ranked, based on the info we have.
+
+  // Select nearest surface as base
   const scores: any = [];
-  let isAcceptable: boolean;
   let ci: number = nextIndex;
   for (; ci < surfaces.length; ci++) {
     candidate = surfaces[ci];
     let score: number = 0;
-    isAcceptable = true;
+    let count: number = 0;
     for (let x = 0; x < top.values.length; x++) {
-      if (top.values[x] !== null && candidate.values[x] === null) {
-        isAcceptable = false;
-        break;
-      } else {
-        score += 1;
+      if (top.values[x] === null || candidate.values[x] === null) {
+        continue;
       }
+      score += candidate.values[x] - top.values[x];
```

Comment:
I admit I don't fully understand what is the goal of this function, I think it is looking for the index of the surface that is closest to `top` (whatever that is). If my interpretation is correct, shouldn't we take the absolute value of the score, or maybe square it? Otherwise, oscillations can cancel each other out (I know it's unlikely to happen, but better safe than sorry)

Reply:
Need signed as surfaces could be above the referenced surface (unlikely).